### PR TITLE
Toggle button

### PR DESCRIPTION
**Pull request by murilopereirame**
Creates new Toggle component

### CODE REVIEW

File: src/common/gui/base/Toggle.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		const childrenArr = [children, this.buildTogglePillComponent(onclick, disabled)]
+		if (togglePosition === "left") {
+			childrenArr.reverse()
+		}
+
+		return m(
+			"label.toggle",
+			{
+				class: this.resolveClasses(disabled, variant),
+				role: "checkbox",
```
Comment (rih-tutao) [inline src/common/gui/base/Toggle.ts:52]: > Toggle button: A two-state button that can be either off (not pressed) or on (pressed). To tell assistive technologies that a button is a toggle button, specify a value for the attribute [aria-pressed](https://w3c.github.io/aria/#aria-pressed). For example, a button labelled mute in an audio player could indicate that sound is muted by setting the pressed state true. Important: it is critical the label on a toggle does not change when its state changes. In this example, when the pressed state is true, the label remains "Mute" so a screen reader would say something like "Mute toggle button pressed". Alternatively, if the design were to call for the button label to change from "Mute" to "Unmute," the aria-pressed attribute would not be needed.

- https://www.w3.org/WAI/ARIA/apg/patterns/button/

Should we do this as an Aria Button instead?

File: src/common/gui/base/Toggle.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import m, { ClassComponent, Vnode } from "mithril"
+import { Keys, TabIndex } from "../../api/common/TutanotaConstants.js"
+import { isKeyPressed } from "../../misc/KeyManager.js"
+
+type ToggleVariant = "normal" | "expanded"
```
Comment (rih-tutao) [inline src/common/gui/base/Toggle.ts:5]: Does an expanded button take up the whole line?

File: src/common/gui/base/Toggle.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+				class: this.checkboxDom?.checked ? "checked" : "unchecked",
+			},
+			m("input[type='checkbox']", {
+				onclick: () => {
+					onclick(this.checkboxDom?.checked ?? false)
+				},
+				oncreate: ({ dom }) => {
+					this.checkboxDom = dom as HTMLInputElement
+				},
+				tabIndex: TabIndex.Programmatic,
```
Comment (rih-tutao) [inline src/common/gui/base/Toggle.ts:81]: Just for clarity, we set the checkbox's tab index to unavailable, but we can tab to the parent and do the logic there, right?

**Review by rih-tutao — CHANGES_REQUESTED** [review]
Looks good! But we might want to use a `toggle button` ARIA pattern (see code comment)

File: src/common/gui/base/Toggle.ts (hunk 2 above)
Comment (murilopereirame) [reply src/common/gui/base/Toggle.ts:5]: Yes

File: src/common/gui/base/Toggle.ts (hunk 3 above)
Comment (murilopereirame) [reply src/common/gui/base/Toggle.ts:81]: Yes, the checkbox is to transfer the state update to the browser. Then we just listen to changes there

File: src/common/gui/base/Toggle.ts (hunk 1 above)
Comment (murilopereirame) [reply src/common/gui/base/Toggle.ts]: Actually, we should use this one https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/

**Review by rih-tutao — APPROVED** [review]
Feels very satisfying to use

**Pull request merged**